### PR TITLE
Update thermoregulation.dm

### LINKED ISSUE #9104: Diona Nymphs freeze on ventcrawl
@DrCelt
@DrCelt
@DrCelt

Even when crawling into a 20C pipe filled with air, they freeze nearly instantly
This also affects monkeys and other vent crawling entities, such as xenos.

"According to Celt this is a feature because more air pressure = more heat transfer so you freeze to 20c ;)"

Last issue was locked and I cannot imagine why.

### PR DESCRIPTION
**Pull request by ghost**
0% tested but this is legit the way to fix this. Will test tomorrow, did this in git's thingy because intigracy won't stop annoying me it's midnight and I'm not on my computer

should work probably but I will remind you this isn't tested
fixes #9104

### CODE REVIEW

**Comment by PJB3005** [comment]
Okay sorry.

This is such a crappy fix you fucking moron.

THAT'S NOT HOW THERMAL TRANSFER WORKS.

**Comment by clusterfack** [comment]
it would be better to check is_ventcrawling I think

**Comment by ghost** [comment]
@PJB3005 
What do you mean?
@clusterfack 
Is that a variable? What a waste.

**Comment by Boxe** [comment]
>0% tested

DNM tag where

**Comment by ghost** [comment]
To be honest if someone merges it after me mentioning 3 times in the OP it hasn't been tested, a DNM tag isn't going to stop them

**Comment by Shadowmech88** [comment]
I'm asking honestly, because I don't know.
Are you sure that higher pressures would realistically cause more heat loss to the environment, and would cause enough heat loss for a person to freeze to death in 20C air?

**Comment by clusterfack** [comment]
I believe the rate at which heat is convected is proportional to the temperature difference to the fourth power. 

So it is rather unlikely that 20C would be enough to kill you in reality.

**Comment by ghost** [comment]
Convected, @Clusterfack , not conducted.
@shadowmech88 You should actually be killed by the pressure first. However, when there are more moles in the atmosphere, there is more heat being transferred by conduction. (Due to greater density) This is the same reason why you lose heat faster in water of 10°C than air of 10°C - you actually lose heat 25 times faster in water than in air. In technicality, I should be measuring heat capacity rather than pressure as a metric, though, but the overall effect would be the same.

**Comment by clusterfack** [comment]
??? It always said convected

**Comment by ghost** [comment]
No, what I meant is that's just convection.

**Comment by clusterfack** [comment]
Yes and convection applies here since we have a solid and a gas and not two solids

**Comment by FudgePucker** [comment]
AWW SHIT IS CELT TRYING TO OUT AUTISM CLUSTER?
OH SHIT PLACE YOUR BETS RIGHT HERE FOAKS, THE ODDS ARE 20:1 IN CLUSTERS FAVOR.
SELLING TICKETS FOR THE MATCH OF THE CENTURY.

**Comment by ghost** [comment]
@clusterfack >humans only lose heat via convection
But that's wrong you nerd, they lose it by both conduction and convection. Where the hell did you read humans only lose heat via convection?

**Comment by ghost** [comment]
@FudgePucker I'll bet 200 pomf coins on myself.

**Comment by gbasood** [comment]
he actually did a code thing again
absolute madman

**Comment by clusterfack** [comment]
You fucking retard even if they are conducting heat into their clothes, their clothes are losing heat through convection. So the limiting factor in all cases is the convection.

**Comment by gbasood** [comment]
Yeah there's not a lot of instances we'd want to model conduction in spess

**Comment by ghost** [comment]
What in Hell's name are you talking about? Even a naked person conducts heat away from them, through air conduction, which increases massively when there is a higher heat capacity / pressure.

**Comment by clusterfack** [comment]
Holy fucking shit dude that is CONVECTION

**Comment by ghost** [comment]
No it isn't. Conduction transfers heat via direct molecular collision.Convection
When a fluid, such as air or a liquid, is heated and then travels away from the source, it carries the thermal energy along. This type of heat transfer is called convection.

**Comment by clusterfack** [comment]
Conduction doesn't transfer via collision, it transfers via vibrations to adjacent atoms in a solid.

It is convection that transfers via collision.

**Comment by ghost** [comment]
Regardless of whether or not it's conduction or convection that's transferring heat by collision, we know that with a greater density, more heat will be transferred. This is one of the reasons why water is 25 times better at conducting heat away from humans than air. 

> For fluids (such as air or water), higher density typically means higher volumetric heat capacity, which typically means higher heat transfer per volume of fluid convected across the object surface.

**Comment by gbasood** [comment]
So since you're modelling heat transfer based on that does that mean dionae nymphs will burn alive in high-pressure pipes that contain room temperature gasses at high pressure?

**Comment by ghost** [comment]
@gbasood No, and I'm not sure why you would think that. They would lose heat slower however if they were in a low temperature pipe. (Or would if people didn't make space cold.)

**Comment by gbasood** [comment]
Because heat transfer code has so far been CRAZY and I just want to make sure it's not THAT crazy

**Comment by gbasood** [comment]
Consensus says make it better

File: code/modules/mob/thermoregulation.dm [hunk 1]
```diff
@@ -65,7 +65,7 @@
 	if(loc_temp < bodytemperature)
 		// We're going to try and just use exposed area(temperature difference)/cold divisor, and assume we're only conducting.
 		var/thermal_loss = (1-get_cold_protection())  				// How much of your skin is exposed.
-		if(environment.total_moles > MOLES_CELLSTANDARD || !IS_SPACE_COLD)
+		if((environment.total_moles > MOLES_CELLSTANDARD || !IS_SPACE_COLD)&&(!istype(src.loc,/obj/machinery/atmospherics)))
 			thermal_loss	*= environment.total_moles/MOLES_CELLSTANDARD	// Multiplied by how many moles are in the environment over 103.934, the normal value of a station. - More moles means more heat transfer, that's basic science.
```
Comment (9600bauds) [inline code/modules/mob/thermoregulation.dm:69]: > More moles means more heat transfer, that's basic science.

So wouldn't less moles mean less heat transfer?

Comment (clusterfack) [inline code/modules/mob/thermoregulation.dm]: FEWER

Comment (clusterfack) [inline code/modules/mob/thermoregulation.dm]: Also we should probably just remove this, and pretend that human bodies just convect heat at a rate that depends on the temperature differential and surface area alone

Comment (9600bauds) [inline code/modules/mob/thermoregulation.dm:69]: > FEWER

So wouldn't fewer moles mean fewer heat transfer?

Comment (clusterfack) [inline code/modules/mob/thermoregulation.dm]: Now you're doing it on purpose
